components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/ChallengeQuestionManager.java

```diff
@@ -550,12 +551,18 @@ public ChallengeQuestion getUserChallengeQuestion(User user, String challengesUr
     private Map<String, String> retrieveAnsweredChallenges(User user, UserChallengeAnswer[] userChallengeAnswers)
             throws IdentityRecoveryException {
 
-        List<String> claimsList = new ArrayList<>();
-        for (UserChallengeAnswer answer : userChallengeAnswers) {
-            claimsList.add(answer.getQuestion().getQuestionSetId().trim());
+        Map<String, String> existingQuestionAndAnswers;
```

Comment:
Can initialize to a empty map and avoid the else block

Reply:
Fixed https://github.com/wso2-extensions/identity-governance/pull/426/commits/4a0c9906dd239c11ee1452cd66195e0c0d924b25